src/transformers/models/llama/modeling_llama.py

```diff
@@ -1061,9 +1061,10 @@ def _update_causal_mask(self, attention_mask, input_tensor):
         device = input_tensor.device
 
         # support going beyond cached `max_position_embedding`
-        if seq_length > self.causal_mask.shape[-1]:
-            causal_mask = torch.full((2 * self.causal_mask.shape[-1], 2 * self.causal_mask.shape[-1]), fill_value=1)
-            self.register_buffer("causal_mask", torch.triu(causal_mask, diagonal=1), persistent=False)
+        causal_mask = self.causal_mask
+        while seq_length > causal_mask.shape[-1]:
+            causal_mask = torch.full((2 * causal_mask.shape[-1], 2 * causal_mask.shape[-1]), fill_value=1)
```

Comment:
```suggestion
        while seq_length > causal_mask.shape[-1]:
            new_max_positions = round(seq_length / causal_mask.shape[-1]) * causal_mask.shape[-1]
            causal_mask = torch.full((new_max_positions, new_max_positions), fill_value=1)
```

Reply:
this should offer a tradeoff between loading too big of a mask and increasing it based on the length of the input